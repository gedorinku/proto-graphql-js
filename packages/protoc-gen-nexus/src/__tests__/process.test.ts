import {
  itGeneratesNexusDSLsToMatchSnapshtos,
  generateDSLs,
  snapshotGeneratedFiles,
} from "./__helpers__/process.test.helper";

describe("simple proto file", () => {
  itGeneratesNexusDSLsToMatchSnapshtos("primitives", [
    "primitives/primitives_pb_nexus.ts",
  ]);

  it("generates nexus DSLs with graphql_type file layout", async () => {
    const resp = await generateDSLs("primitives", "protobufjs", {
      perGraphQLType: true,
    });
    snapshotGeneratedFiles(resp, [
      "primitives/Message.nexus.ts",
      "primitives/Primitives.nexus.ts",
      "primitives/MessageInput.nexus.ts",
      "primitives/PrimitivesInput.nexus.ts",
    ]);
  });

  it("generates nexus DSLs with partial inputs", async () => {
    const resp = await generateDSLs("primitives", "protobufjs", {
      partialInputs: true,
    });
    snapshotGeneratedFiles(resp, ["primitives/primitives_pb_nexus.ts"]);
  });
});

describe("well-known protobuf types", () => {
  itGeneratesNexusDSLsToMatchSnapshtos("wktypes", [
    "wktypes/well_known_types_pb_nexus.ts",
  ]);
});

describe("protobuf enums", () => {
  itGeneratesNexusDSLsToMatchSnapshtos("enums", ["enums/enums_pb_nexus.ts"]);

  it("generates nexus DSLs with graphql_type file layout", async () => {
    const resp = await generateDSLs("enums", "protobufjs", {
      perGraphQLType: true,
    });
    snapshotGeneratedFiles(resp, [
      "enums/MessageWithEnums.nexus.ts",
      "enums/MyEnum.nexus.ts",
      "enums/MyEnumWithoutUnspecified.nexus.ts",
      "enums/MessageWithEnumsInput.nexus.ts",
    ]);
  });
});

describe("nested protobuf types", () => {
  itGeneratesNexusDSLsToMatchSnapshtos("nested", ["nested/nested_pb_nexus.ts"]);
});

describe("protobuf custom options", () => {
  itGeneratesNexusDSLsToMatchSnapshtos("extensions", [
    "extensions/extensions_pb_nexus.ts",
    "extensions/ignored_pb_nexus.ts",
  ]);

  it("generates nexus DSLs with graphql_type file layout", async () => {
    const resp = await generateDSLs("extensions", "protobufjs", {
      perGraphQLType: true,
    });
    snapshotGeneratedFiles(resp, [
      "extensions/TestPrefixPrefixedMessage.nexus.ts",
      "extensions/TestPrefixPrefixedMessageInnerMessage.nexus.ts",
      "extensions/TestPrefixPrefixedMessageInnerMessage2.nexus.ts",
      "extensions/TestPrefixPrefixedMessageSquashedMessage.nexus.ts",
      "extensions/TestPrefixPrefixedMessagePartialIgnoreOneof.nexus.ts",
      "extensions/TestPrefixPrefixedEnum.nexus.ts",
      "extensions/TestPrefixIgnoredMessageNotIgnored.nexus.ts",
      "extensions/TestPrefixInterfaceMessage.nexus.ts",
      "extensions/TestPrefixInterfaceMessageType.nexus.ts",
      "extensions/TestPrefixRenamedMessage.nexus.ts",
      "extensions/TestPrefixPrefixedMessageInput.nexus.ts",
      "extensions/TestPrefixPrefixedMessageInnerMessageInput.nexus.ts",
      "extensions/TestPrefixPrefixedMessageInnerMessage2Input.nexus.ts",
      "extensions/TestPrefixPrefixedMessageSquashedMessageInput.nexus.ts",
      "extensions/TestPrefixIgnoredMessageNotIgnoredInput.nexus.ts",
      "extensions/TestPrefixInterfaceMessageInput.nexus.ts",
      "extensions/TestPrefixRenamedMessageInput.nexus.ts",
    ]);
  });
});

describe("protobuf oneof", () => {
  itGeneratesNexusDSLsToMatchSnapshtos("oneof", ["oneof/oneof_pb_nexus.ts"]);

  it("generates nexus DSLs with graphql_type file layout", async () => {
    const resp = await generateDSLs("oneof", "protobufjs", {
      perGraphQLType: true,
    });
    snapshotGeneratedFiles(resp, [
      "oneof/OneofParent.nexus.ts",
      "oneof/OneofParentRequiredOneofMembers.nexus.ts",
      "oneof/OneofParentOptionalOneofMembers.nexus.ts",
      "oneof/OneofMemberMessage1.nexus.ts",
      "oneof/OneofMemberMessage2.nexus.ts",
      "oneof/OneofParentInput.nexus.ts",
      "oneof/OneofMemberMessage1Input.nexus.ts",
      "oneof/OneofMemberMessage2Input.nexus.ts",
    ]);
  });
});

describe("multipkgs", () => {
  itGeneratesNexusDSLsToMatchSnapshtos("multipkgs/subpkg1", [
    "multipkgs/subpkg1/types_pb_nexus.ts",
  ]);
  itGeneratesNexusDSLsToMatchSnapshtos("multipkgs/subpkg2", [
    "multipkgs/subpkg2/types_pb_nexus.ts",
  ]);
});

describe("deprecation", () => {
  itGeneratesNexusDSLsToMatchSnapshtos("deprecation", [
    "deprecation/deprecation_pb_nexus.ts",
    "deprecation/file_deprecation_pb_nexus.ts",
  ]);
});

describe("field_behavior", () => {
  itGeneratesNexusDSLsToMatchSnapshtos("field_behavior", [
    "field_behavior/comments_pb_nexus.ts",
  ]);
});

describe("empty types", () => {
  itGeneratesNexusDSLsToMatchSnapshtos("empty_types", [
    "empty_types/empty_pb_nexus.ts",
  ]);
});
