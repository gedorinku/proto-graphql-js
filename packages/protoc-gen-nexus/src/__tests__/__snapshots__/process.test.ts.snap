// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates nexus DSL from nested proto types 1`] = `
"// Code generated by protoc-gen-nexus. DO NOT EDIT.
// source: nested/nested.proto

import { objectType } from \\"@nexus/schema\\";
import * as _$nested$nested_pb from \\"./nested/nested_pb\\";
export type _$nested$nested_pb$ParentMessage = _$nested$nested_pb.ParentMessage;
export type _$nested$nested_pb$ParentMessageNestedMessage = _$nested$nested_pb.ParentMessage.NestedMessage;
export const ParentMessage = objectType({
    name: \\"ParentMessage\\",
    description: \\"\\",
    definition(t) {
        t.nullable.string(\\"body\\", {
            description: \\"\\",
            resolve(root) { return root.getBody(); }
        });
        t.nullable.field(\\"nested\\", {
            description: \\"\\",
            type: \\"NestedMessage\\",
            resolve(root) { return root.getNested(); }
        });
    },
    sourceType: { module: __filename, export: \\"_$nested$nested_pb$ParentMessage\\" }
});
export const ParentMessageNestedMessage = objectType({
    name: \\"NestedMessage\\",
    description: \\"\\",
    definition(t) {
        t.nullable.string(\\"nestedBody\\", {
            description: \\"\\",
            resolve(root) { return root.getNestedBody(); }
        });
    },
    sourceType: { module: __filename, export: \\"_$nested$nested_pb$ParentMessageNestedMessage\\" }
});
"
`;

exports[`generates nexus DSL from proto well-known types 1`] = `
"// Code generated by protoc-gen-nexus. DO NOT EDIT.
// source: wktypes/well_known_types.proto

import { objectType } from \\"@nexus/schema\\";
import * as proto_nexus from \\"proto-nexus\\";
import * as _$wktypes$well_known_types_pb from \\"./wktypes/well_known_types_pb\\";
export type _$wktypes$well_known_types_pb$Message = _$wktypes$well_known_types_pb.Message;
export const Message = objectType({
    name: \\"Message\\",
    description: \\"\\",
    definition(t) {
        t.nullable.dateTime(\\"timestamp\\", {
            description: \\"\\",
            resolve(root) { return proto_nexus.timestampToDate(root.getTimestamp()); }
        });
        t.nullable.int(\\"int32Value\\", {
            description: \\"\\",
            resolve(root) { return proto_nexus.unwrapInt32Value(root.getInt32Value()); }
        });
        t.nullable.int(\\"int64Value\\", {
            description: \\"\\",
            resolve(root) { return proto_nexus.unwrapInt64Value(root.getInt64Value()); }
        });
        t.nullable.int(\\"uint32Value\\", {
            description: \\"\\",
            resolve(root) { return proto_nexus.unwrapUInt32Value(root.getUint32Value()); }
        });
        t.nullable.int(\\"uint64Value\\", {
            description: \\"\\",
            resolve(root) { return proto_nexus.unwrapUInt64Value(root.getUint64Value()); }
        });
        t.nullable.float(\\"floatValue\\", {
            description: \\"\\",
            resolve(root) { return proto_nexus.unwrapFloatValue(root.getFloatValue()); }
        });
        t.nullable.float(\\"doubleValue\\", {
            description: \\"\\",
            resolve(root) { return proto_nexus.unwrapDoubleValue(root.getDoubleValue()); }
        });
        t.nullable.boolean(\\"boolValue\\", {
            description: \\"\\",
            resolve(root) { return proto_nexus.unwrapBoolValue(root.getBoolValue()); }
        });
        t.nullable.string(\\"stringValue\\", {
            description: \\"\\",
            resolve(root) { return proto_nexus.unwrapStringValue(root.getStringValue()); }
        });
    },
    sourceType: { module: __filename, export: \\"_$wktypes$well_known_types_pb$Message\\" }
});
"
`;

exports[`generates nexus DSL from simple proto file 1`] = `
"// Code generated by protoc-gen-nexus. DO NOT EDIT.
// source: hello/hello.proto

import { objectType } from \\"@nexus/schema\\";
import * as proto_nexus from \\"proto-nexus\\";
import * as _$hello$hello_pb from \\"./hello/hello_pb\\";
export type _$hello$hello_pb$User = _$hello$hello_pb.User;
export type _$hello$hello_pb$Post = _$hello$hello_pb.Post;
export const User = objectType({
    name: \\"User\\",
    description: \\"User entity\\",
    definition(t) {
        t.nonNull.int(\\"id\\", {
            description: \\"Required. Output only. ID.\\",
            resolve(root) { return root.getId(); }
        });
        t.nonNull.string(\\"name\\", {
            description: \\"Required. User's login name.\\",
            resolve(root) { return root.getName(); }
        });
        t.nonNull.list.field(\\"posts\\", {
            description: \\"Required. Posts that are written by the user.\\",
            type: \\"Post\\",
            resolve(root) { return root.getPostsList(); }
        });
    },
    sourceType: { module: __filename, export: \\"_$hello$hello_pb$User\\" }
});
export const Post = objectType({
    name: \\"Post\\",
    description: \\"Post entity\\",
    definition(t) {
        t.nonNull.int(\\"id\\", {
            description: \\"Required. Output only. ID.\\",
            resolve(root) { return root.getId(); }
        });
        t.nonNull.string(\\"title\\", {
            description: \\"Required. Article.\\",
            resolve(root) { return root.getTitle(); }
        });
        t.nonNull.string(\\"body\\", {
            description: \\"Required. Body.\\",
            resolve(root) { return root.getBody(); }
        });
        t.nonNull.dateTime(\\"publishedTime\\", {
            description: \\"Required. Output only. Published time.\\",
            resolve(root) { return proto_nexus.timestampToDate(root.getPublishedTime()); }
        });
    },
    sourceType: { module: __filename, export: \\"_$hello$hello_pb$Post\\" }
});
"
`;
